### PR TITLE
Persist location

### LINKED ISSUE #93: Bio/Profile onboarding screen flashes always
ezgif-5-a61891dd065e

Even though I already have an account, this always happens when I open Buddies.

cc @JakeThurman Without really looking, I'm guessing is related to #90 which fixed #18

### PR DESCRIPTION
**Pull request by JakeThurman**
rebased on top of #100 

Just the top 2 commits:

```
ec0156f Add Location Persistence handler (Jake Thurman)
8bcb21b Fix #93 - Bio screen always flashes (Jake Thurman)
```

To test:
- Open the app & login
    - Observe while opening that you didn't see the bio screen flash (wow)
- Observe that your location is now San Fransisco. (setup as simulator default)
    - In firebase you will see this as: [37.7873589, -122.408227]

### CODE REVIEW

File: Buddies/Utilities/LocationPersistence.swift [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    init(dataAccessor: DataAccessor = DataAccessor.instance) {
+        super.init()
+        
+        manager.desiredAccuracy = kCLLocationAccuracyBest
+        manager.delegate = self
+        
+        cancelUserListener = dataAccessor.useLoggedInUser { user in
+            self.user = user
+            
+            // On user change, record where we are imediately
```
Comment (noahtallen) [inline Buddies/Utilities/LocationPersistence.swift]: This seems very backwards - shouldn't it (and the code) be "on location change, record where we are immediately"?

File: Buddies/Utilities/LocationPersistence.swift [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    }
+    
+    func makeSureWeHaveLocationAccess(from vc: UIViewController) {
+        let status = CLLocationManager.authorizationStatus()
+        
+        switch status {
+        case .notDetermined:
+            manager.requestWhenInUseAuthorization()
+        case .denied, .restricted:
+            let alert = UIAlertController(title: "Location Services disabled", message: "Please enable Location Services for Buddies in Settings", preferredStyle: .alert)
```
Comment (noahtallen) [inline Buddies/Utilities/LocationPersistence.swift:47]: I could see this getting super annoying. Are we going to force users to use location to use the app like Zonder does?

File: Buddies/Onboarding/SignUpInfoVC.swift [hunk 3]
```diff
@@ -188,7 +190,7 @@ class SignUpInfoVC: LoginBase, UIImagePickerControllerDelegate, UINavigationCont
             let blockedActivities: [String] = []
             let blockedBy: [String] = []
             let dateJoined = Timestamp(date: Date())
-            let loc = GeoPoint.init(latitude: 10, longitude: 10)
+            let loc = GeoPoint(latitude: 0, longitude: 0)//this should get replaced momentatily
```
Comment (noahtallen) [inline Buddies/Onboarding/SignUpInfoVC.swift]: why even initialize it to 0, 0? If the user denies location, shouldn't we store "null" or something more clear like that?

**Review by noahtallen — CHANGES_REQUESTED** [review]
i feel like we could make some improvements to this one, but it's a great start!

File: Buddies/Onboarding/SignUpInfoVC.swift [hunk 4]
```diff
@@ -68,6 +68,8 @@ class SignUpInfoVC: LoginBase, UIImagePickerControllerDelegate, UINavigationCont
         bioText.layer.borderColor = ControlColors.fieldBorderFocused.cgColor
         bioText.layer.borderWidth = 2
         buttonPicture.layer.cornerRadius = buttonPicture.frame.size.width / 2
+        
+        LocationPersistence.instance.makeSureWeHaveLocationAccess(from: self)
```
Comment (noahtallen) [inline Buddies/Onboarding/SignUpInfoVC.swift:76]: another quick note here, I _think_ that this popup is going to "conflict" with the notification request. It's not a big deal - it just feels bad.

File: Buddies/Onboarding/SignUpInfoVC.swift (hunk 3 above)
Comment (lukemeierj) [reply Buddies/Onboarding/SignUpInfoVC.swift]: Just a spelling nit-pick spelling: "momentarily" ;)

File: Buddies/Utilities/LocationPersistence.swift (hunk 1 above)
Comment (JakeThurman) [reply Buddies/Utilities/LocationPersistence.swift]: It's also watched below when moving

Comment (JakeThurman) [reply Buddies/Utilities/LocationPersistence.swift]: this for onlogin

File: Buddies/Utilities/LocationPersistence.swift (hunk 2 above)
Comment (JakeThurman) [reply Buddies/Utilities/LocationPersistence.swift:47]: yes

**Comment by JakeThurman** [comment]
@noahtallen I think I fixed it (hopefully for real this time?) lol

**Comment by JakeThurman** [comment]
Rebased

**Review by noahtallen — APPROVED** [review]

**Pull request merged**